prusti-interface/src/prusti_error.rs

```diff
@@ -100,6 +130,7 @@ impl PrustiError {
 
     /// Report the encoding error using the compiler's interface
     pub fn emit(self, env: &Environment) {
+        assert!(!self.is_disabled);
```

Comment:
Why not add the logic for skipping error emission here into `emit`, instead of requiring the caller to use `emit` or `cancel`?

Reply:
My intention was to allow disabling errors only in places where the caller expects them. Otherwise, we may end up with bugs that silently swallow errors.